contracts/PangolinExchange.sol

```diff
@@ -48,18 +48,24 @@ contract PangolinExchange is Ownable, IAssetsExchange {
    * @dev _token ERC20 token's address
    * @dev _exactERC20AmountOut amount of the ERC20 token to be bought
   **/
-  function buyAsset(bytes32 _token, uint256 _exactERC20AmountOut) payable external override RefundRemainder {
+  function buyAsset(bytes32 _token, uint256 _exactERC20AmountOut) payable external override returns(bool){
     require(_exactERC20AmountOut > 0, "Amount of tokens to buy has to be greater than 0");
-
     address tokenAddress = supportedAssets.getAssetAddress(_token);
-
     uint256 amountIn = getEstimatedAVAXForERC20Token(_exactERC20AmountOut, tokenAddress);
-
     require(msg.value >= amountIn, "Not enough funds provided");
 
-    pangolinRouter.swapAVAXForExactTokens{value : msg.value}(_exactERC20AmountOut, getPathForAVAXtoToken(tokenAddress), msg.sender, block.timestamp);
+//    pangolinRouter.swapAVAXForExactTokens{value : msg.value}(_exactERC20AmountOut, getPathForAVAXtoToken(tokenAddress), msg.sender, block.timestamp);
 
+    (bool success, bytes memory returnData) = address(pangolinRouter).call{value : msg.value}(abi.encodeWithSignature("swapAVAXForExactTokens(uint,address[],address,uint)", _exactERC20AmountOut, getPathForAVAXtoToken(tokenAddress), msg.sender, block.timestamp));
```

Comment:
`uint256`

Reply:
This wasn't ready for a review at that point. It was there just to showcase the current state of code that didn't work as expected. It is fixed now.